packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -183,6 +189,198 @@ export const getAllLatestEntitiesResolver: ResolverFn<
   return entitySubgraph as Subgraph;
 };
 
+export const aggregateEntitiesResolver: Extract<
+  QueryResolvers<LoggedInGraphQLContext>["aggregateEntities"],
+  Function
+> = async (
+  _,
+  {
+    operation,
+    constrainsValuesOn,
+    constrainsPropertiesOn,
+    constrainsLinksOn,
+    constrainsLinkDestinationsOn,
+    isOfType,
+    hasLeftEntity,
+    hasRightEntity,
+  },
+  { dataSources },
+  __,
+) => {
+  const { graphApi } = dataSources;
+
+  /** @todo - should we throw an error, or should we just ignore the constraint and hope for the best */
+  if (
+    (operation.pageNumber !== undefined && operation.pageNumber !== null) ||
+    (operation.itemsPerPage !== undefined && operation.itemsPerPage !== null)
+  ) {
+    throw new ApolloError(
+      "Pagination on aggregateEntities results is not currently supported",
+    );
+  }
+
+  if (operation.multiSort !== undefined && operation.multiSort !== null) {
+    throw new ApolloError(
+      "Sorting on aggregateEntities results is not currently supported",
+    );
+  }
+
+  /** @todo - Encapsulate this transpilation in a shared function somewhere */
+  const filter: Filter = {
+    all: [
+      {
+        equal: [{ path: ["version"] }, { parameter: "latest" }],
+      },
+      {
+        equal: [{ path: ["archived"] }, { parameter: false }],
+      },
+    ],
+  };
+
+  if (operation.entityTypeId) {
+    filter.all.push({
+      equal: [
+        { path: ["type", "versionedUri"] },
+        { parameter: operation.entityTypeId },
+      ],
+    });
+  }
+
+  if (operation.multiFilter !== undefined && operation.multiFilter !== null) {
+    const subFilters: Filter[] = [];
+
+    /**
+     * @todo - support more than just "properties" as a field so we can filter on other things in the entity
+     *   https://app.asana.com/0/0/1203749694686236/f
+     */
```

Comment:
Right now there is no way to filter based on metadata (such as link destinations, order, timestamps, etc.)

Reply:
Where does support for this need to be added? the BP, SQ, or both?

Reply:
SQ supports almost all of those (except nested fields in `properties`), the BP could support it if we just took an array of fields and said in the spec that it represents a path within the `Entity` rather than `entity.properties`. 

That would mean that _any_ field we add inside the `Entity` in BP, we'd have to add native support for in SQ, and we'd have to check if there were any that we hadn't done that for yet (there are _maybe_ some metadata fields that we didn't bother with yet)

Reply:
> the BP could support it if we just took an array of fields and said in the spec that it represents a path within the Entity rather than entity.properties.

I think we should do this for `0.3` – mainly thinking of the current special treatment of `entityTypeId` as a bit weird and undesirable.

> That would mean that any field we add inside the Entity in BP, we'd have to add native support for in SQ, and we'd have to check if there were any that we hadn't done that for yet (there are maybe some metadata fields that we didn't bother with yet)

I think we'll want this anyway (though full support in HASH can lag behind a bit)

Reply:
This was rectified